src/NexusMods.App.UI/Overlays/Updater/UpdaterView.axaml.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using System.Windows.Input;
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Markup.Xaml;
+using Avalonia.ReactiveUI;
+using ReactiveUI;
+
+namespace NexusMods.App.UI.Overlays.Updater;
+
+public partial class UpdaterView : ReactiveUserControl<IUpdaterViewModel>
+{
+    public UpdaterView()
+    {
+        InitializeComponent();
+
+        this.WhenActivated(d =>
+        {
+            this.WhenAnyValue(view => view.ViewModel!.OldVersion)
+                .Select(v => $"v{v}")
+                .BindToUi(this, view => view.FromVersionTextBlock.Text)
+                .DisposeWith(d);
+
+            this.WhenAnyValue(view => view.ViewModel!.NewVersion)
+                .Select(v => $"v{v}")
+                .BindToUi(this, view => view.ToVersionTextBlock.Text)
+                .DisposeWith(d);
+
+            this.WhenAnyValue(view => view.ViewModel!.ShowSystemUpdateMessage)
+                .Select(show => !show)
+                .BindToUi(this, view => view.UpdateButton.IsEnabled)
+                .DisposeWith(d);
+
+            this.WhenAnyValue(view => view.ViewModel!.ShowSystemUpdateMessage)
+                .BindToUi(this, view => view.UseSystemUpdater.IsVisible)
+                .DisposeWith(d);
```

Comment:
These should use `OneWayBind` instead of `BindToUi`. That method also has an optional lambda argument to transform the inputs before they are passed to the component.

Reply:
Unfortunately this binds on whatever thread fires the event in the VM, so it'll throw errors if *anything* modifies the VM but isn't on the UI thread.